commercetools.Sdk/commercetools.Sdk.Client/QueryCommandExtensions.cs

```diff
@@ -0,0 +1,198 @@
+                queryParameters.Expand.Add(expression);
+            }
+
+            return command;
+        }
+
+        public static QueryCommand<T> Limit<T>(this QueryCommand<T> command, int limit)
+        {
+            if (command.QueryParameters is IPageable queryParameters)
+            {
+                queryParameters.Limit = limit;
+            }
+
+            return command;
+        }
+
+        public static QueryCommand<T> Offset<T>(this QueryCommand<T> command, int offset)
+        {
+            if (command.QueryParameters is IPageable queryParameters)
+            {
+                queryParameters.Offset = offset;
+            }
+
+            return command;
+        }
+
+        public static QueryCommand<T> WithTotal<T>(this QueryCommand<T> command, bool withTotal)
+        {
+            if (command.QueryParameters is IPageable queryParameters)
+            {
+                queryParameters.WithTotal = withTotal;
+            }
+
+            return command;
+        }
+
+        public static QueryCommand<T> Sort<T>(this QueryCommand<T> command, string expression)
+        {
+            if (command.QueryParameters is ISortable queryParameters)
+            {
+                queryParameters.Sort.Add(expression);
+            }
+
+            return command;
+        }
+
+        public static QueryCommand<T> Sort<T>(this QueryCommand<T> command, Expression<Func<T, IComparable>> expression, SortDirection sortDirection = SortDirection.Ascending)
+        {
+            if (command.QueryParameters is ISortable queryParameters)
+            {
+                queryParameters.Sort.Add(new Sort<T>(expression, sortDirection).ToString());
+            }
+
+            return command;
+        }
+
+        public static QueryCommand<T> SetExpand<T>(this QueryCommand<T> command, IEnumerable<Expansion<T>> expandPredicates)
+        {
+            if (command.QueryParameters is IExpandable queryParameters && expandPredicates != null)
+            {
+                return command.SetExpand(expandPredicates.Select(predicate => predicate.ToString()));
+            }
+
+            return command;
+        }
+
+        public static QueryCommand<T> SetExpand<T>(this QueryCommand<T> command, IEnumerable<string> expandPredicates)
+        {
+            if (command.QueryParameters is IExpandable queryParameters && expandPredicates != null)
+            {
+                queryParameters.Expand.Clear();
+                foreach (var expand in expandPredicates)
+                {
+                    queryParameters.Expand.Add(expand);
+                }
+            }
+
+            return command;
+        }
+
+        public static QueryCommand<T> SetWhere<T>(this QueryCommand<T> command, IEnumerable<QueryPredicate<T>> queryPredicates)
+        {
+            if (command.QueryParameters is IPredicateQueryable queryParameters && queryPredicates != null)
+            {
+                return command.SetWhere(queryPredicates);
```

Comment:
will result in stack overflow as the predicates are never rendered to a string

e.g.: `queryPredicates.Select(predicate => predicate.ToString()))`

Reply:
it's fixed now